airflow/jobs/scheduler_job.py

```diff
@@ -1092,7 +1099,7 @@ def _enqueue_task_instances_with_queued_state(self, simple_dag_bag,
                 ignore_task_deps=False,
                 ignore_ti_state=False,
                 pool=simple_task_instance.pool,
-                file_path=simple_dag.full_filepath,
+                file_path=change_file_path(simple_dag.full_filepath),
```

Comment:
I'm also wondering if it makes more sense to do this change in the processor/parser process rather than here?

Reply:
Yeah it can be done in processor as well. It is done here because this change is specifically when we are enqueuing task and sending it to the workers as the worker need to have its own path of the dag. It might be more relevant that at other places in the scheduler the full_filepath variable should contain full path and not the masked one.

Reply:
This is not the right fix for this issue, and is hacking around the problem by adding another place that is doing this logic.

This goes through https://github.com/apache/airflow/blob/ac42428bf530c259ab1b0dca08458c1ebf49b04a/airflow/models/taskinstance.py#L284-L290 which should already do this substition. Can you find out why it is not anymore?

Reply:
Oh I see why. It's becuase we're calling `TI.generate_command` from a SimpleTaskInstance, not a TI. The work Kaxil and I are doing in AIP-24 might fix this. Your test will be useful though.

/cc @kaxil - if we remove SimpleDag/SimpleTI as part of AIP-24 we should add in the new test below.

Reply:
👍